snips_nlu/entity_parser/custom_entity_parser.py

```diff
@@ -123,24 +128,54 @@ def _merge_entity_utterances(raw_utterances, stemmed_utterances):
     return raw_utterances
 
 
-def _create_custom_entity_parser_configuration(entities):
-    return {
-        "entity_parsers": [
-            {
-                "entity_identifier": entity_name,
-                "entity_parser": {
-                    "threshold": entity[MATCHING_STRICTNESS],
-                    "gazetteer": [
-                        {
-                            "raw_value": k,
-                            "resolved_value": v
-                        } for k, v in sorted(iteritems(entity[UTTERANCES]))
-                    ]
-                }
-            } for entity_name, entity in sorted(iteritems(entities))
-        ]
+def _create_custom_entity_parser_configuration(
+        entities, stopwords_fraction, language):
+    """Dynamically creates the gazetteer parser configuration.
+
+    Args:
+        entities (dict): entity for the dataset
+        stopwords_fraction (float): fraction of the vocabulary of
+            the entity values that will be considered as stop words (
+            the top n_vocabulary * stopwords_fraction most frequent words will
+            be considered stop words)
+        language (str): language of the entities
+
+    Returns: the parser configuration as dictionary
+    """
+
+    if not 0 < stopwords_fraction < 1:
+        raise ValueError("stopwords_fraction must be in ]0.0, 1.0[")
+
+    parser_configurations = []
+    for entity_name, entity in sorted(iteritems(entities)):
+        vocabulary = set()
+        entity_tokens = set(
+            t for raw_value in itervalues(entity[UTTERANCES])
+            for t in tokenize_light(raw_value, language)
+        )
+        vocabulary.update(entity_tokens)
```

Comment:
You can directly assign the `set` to `vocabulary` and avoid the `update` op.

Reply:
👍